### DISCUSSION TITLE
Fixing typos in older, already deployed versions

### DISCUSSION THREAD

**Discussion by pmdirector** [jimporter/mike#220] [Q&A] [answered]
From the README: "mike is built around the idea that once you've generated your docs for a particular version, you should never need to touch that version again." I sometimes find typos in older, already deployed versions. I have seen discussions indicating it is possible to fix older versions, but I am unclear as to the easiest way. (I apologize in advance if I missed previous answers.)

Suppose I have deployed versions 1.0 and 2.0, and now I am working on 3.0. The source material has changed quite a bit from 1.0 to 3.0. I find a typo in 1.0. How to fix?
1. Check out `gh-pages` and fix the typos directly in the html files in `1.0/`, or
2. Find the last commit that created the 1.0 source material in the `main` branch. Then create a new branch at that point, fix the typos, and then `mike deploy 1.0` from that branch, or
3. Fix the typos in `main` branch (if the typos are still present). Find the last commit that created the 1.0 source material in the `main` branch. Then create a new branch at that point, cherrypick the typo fixes from `main`, then `mike deploy 1.0` from that branch, or
4. Some other way.

Thanks for any help.

**Comment by jimporter** [comment, marked as answer]
Really, this is just a question about how to work with Git in general, and there are many ways to do that. I can't tell you how to manage the versions of your application (ultimately, do whatever you like). At a high level though, fixing the documentation for an old release is just the same as fixing a code bug. Options 2 or 3 would be perfectly fine depending on how you normally work on your project.

For the mike side of things, yes, all you need to do is `mike deploy 1.0` and it will deploy the contents of your current checkout of your docs to `1.0/*` in your `gh-pages` branch.

**Reply by pmdirector** [reply]
Thanks for the prompt response. I'll go with option 2 because it covers the case where the typo(s) in 1.0 do not appear in later versions.
